extensions/amp-access/0.1/amp-access.js

```diff
@@ -39,18 +39,30 @@ import {xhrFor} from '../../../src/xhr';
 
 /**
  * The configuration properties are:
+ * - type: The type of access workflow: client, server or other.
  * - authorization: The URL of the Authorization endpoint.
  * - pingback: The URL of the Pingback endpoint.
  * - login: The URL of the Login Page.
  *
  * @typedef {{
- *   authorization: string,
- *   pingback: string,
- *   login: string
+ *   type: !AccessType,
+ *   authorization: (string|undefined),
+ *   pingback: (string|undefined),
+ *   login: (string|undefined)
  * }}
  */
 let AccessConfigDef;
 
+/**
+ * The type of access flow.
+ * @enum {string}
+ */
+const AccessType = {
+  CLIENT: 'client',
+  SERVER: 'server',
+  OTHER: 'other',
```

Comment:
The meaning of it is: "I don't know what type of auth this is, so I'm gonna ignore as much as I can". Hence the type is "other". I really didn't find any other better way to short-name it.